src/Paginator/KeysetPaginator.php

```
@@ -49,7 +51,10 @@ public function __construct(DataReaderInterface $dataReader)
     public function read(): iterable
     {
         $this->currentLastValue = null;
-        $dataReader = $this->dataReader->withLimit($this->pageSize);
+        $this->currentFirstValue = null;
+        $isBackwardPagination = $this->pageSize < 0;
```

That looks wrong. Pagesize should not be less than zero. Shouldn't we determine backwards pagination by the presence of `lastValue`?

---

I thought this of paging was relative by point and it became a negative page size for backward paging.
New paging mechanism:
1. if `lastValue` and `firstValue` are null, then next page from begining
2. if `lastValue` is not null, then next page from the specified point
3. if `firstValue` is not null, then previous page from the specified point
4. if `firstValue` and `lastVaule` are not null,  then first line of this listing (1.)

@samdark Is this solution acceptable?

---

> Pagesize should not be less than zero.

Negative page size is fixes.

---

> 1. if lastValue and firstValue are null, then next page from begining

If these both are null then it's meant to be first page.

> 2. if lastValue is not null, then next page from the specified point
> 3. if firstValue is not null, then previous page from the specified point

Correct.

> 4. if firstValue and lastVaule are not null, then first line of this listing (1.)

That's incorrect input. We can't get to previous page and next page at the same time. Either we should throw an exception in this case or use `lastValue` ignoring `firstValue`.

---

> That's incorrect input. We can't get to previous page and next page at the same time. Either we should throw an exception in this case or use lastValue ignoring firstValue.

Sorry about my English, I wanted to write this: `or use lastValue ignoring firstValue.` The code was also implemented on this basis.